gemrb/core/Interface.cpp

```diff
@@ -3349,9 +3357,13 @@ Store *Interface::SetCurrentStore(const ResRef &resName, ieDword owner)
 	}
 
 	CurrentStore = gamedata->GetStore(resName);
-	if (CurrentStore == NULL) {
-		return NULL;
+	if (CurrentStore == nullptr) {
+		vars.Set("BARTER_PC", 0);
+		return nullptr;
 	}
+	// this is set when opening the store and not updated when changing PC
+	// TODO: seems like it would be a good enhancement to set this to the selected PC with the highest charisma
```

Comment:
Yeah, we can do much better — the games always used the party leader, whoever is in the first slot, regardless of who is talking.